### PR TITLE
*: refine code to avoid several unnecessary object allocation

### PR DESCRIPTION
**Pull request by tiancaiamao**
Tiny refactor

### CODE REVIEW

File: executor/aggregate.go [hunk 1]
```diff
@@ -95,7 +96,8 @@ func (e *HashAggExec) Next(goCtx goctx.Context) (Row, error) {
 }
 
 func (e *HashAggExec) getGroupKey(row Row) ([]byte, error) {
-	vals := make([]types.Datum, 0, len(e.GroupByItems))
+	var tmp [32]types.Datum
+	vals := tmp[:]
```
Comment (coocood) [inline executor/aggregate.go]: The `vals` has len 32, you want `vals := tmp[:0]`.
And I think `make([]types.Datum, 0, 32)` is better, the memory is on stack too.

File: server/conn.go [hunk 2]
```diff
@@ -849,7 +849,7 @@ func (cc *clientConn) writeResultset(goCtx goctx.Context, rs ResultSet, binary b
 }
 
 func (cc *clientConn) writeColumnInfo(columns []*ColumnInfo) error {
-	data := make([]byte, 4, 1024)
+	data := cc.alloc.AllocWithLen(4, 1024)
```
Comment (coocood) [inline server/conn.go]: We planned to remove arena allocator.

**Comment by coocood** [comment]
@tiancaiamao 
Check the `OpenTracing.Enable` before start a span in `dispatch`?

File: server/conn.go (hunk 2 above)
Comment (ngaut) [reply server/conn.go]: Why would we remove arena allocator?

Comment (coocood) [reply server/conn.go]: Because we already use append for all the codec, a simple `[]byte` will do the same as arena allocator.

Comment (ngaut) [reply server/conn.go]: The allocator interface is simple and clean. I prefer to use it for now.

File: tidb.go [hunk 3]
```diff
@@ -151,24 +151,25 @@ func Compile(goCtx goctx.Context, ctx context.Context, stmtNode ast.StmtNode) (a
...
+		span, goCtx = opentracing.StartSpanFromContext(goCtx, "runStmt")
+		span.LogKV("sql", s.OriginText())
+		defer span.Finish()
+	}
 
 	var err error
 	var rs ast.RecordSet
 	se := ctx.(*session)
 	rs, err = s.Exec(goCtx)
-	span.SetTag("txn.id", se.sessionVars.TxnCtx.StartTS)
```
Comment (shenli) [inline tidb.go]: This SetTag is missing in the new code.

**Pull request closed without merging**